tensorflow/c/experimental/gradients/BUILD

```diff
@@ -155,6 +155,71 @@ filegroup(
     ],
 )
 
+cc_library(
+    name = "gradient_checker",
+    testonly = 1,
+    srcs = [
+        "gradient_checker.cc",
+    ],
+    hdrs = [
+        "gradient_checker.h",
+    ],
+    visibility = [
+        "//tensorflow:internal",
+    ],
+    deps = [
+        ":unified_api_testutil",
+        "//tensorflow/c/eager:abstract_tensor_handle",
+        "@com_google_absl//absl/types:span",
+        "//tensorflow/c/experimental/ops:math_ops",
+    ] + if_libtpu(
```

Comment:
Since `unified_api_testutil ` does not have `if_libtpu`, this option seems not necessary to me.

Reply:
`//tensorflow/compiler/mlir/tensorflow/c:mlir_c_api_registration` should only be needed on binary and test targets and not on library targets so we can drop this I think.